front/src/common/Map/Search/useSearchOperationalPoint.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState, useEffect, useMemo } from 'react';
+
+import { compact } from 'lodash';
+
+import { type SearchResultItemOperationalPoint, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useInfraID } from 'common/osrdContext';
+import { useDebounce } from 'utils/helpers';
+
+export const mainOperationalPointsCHCodes = ['', '00', 'BV'];
+
+type SearchOperationalPoint = {
+  debounceDelay?: number;
+  mainOperationalPointsOnly?: boolean;
+  searchTerm?: string;
+  chCodeFilter?: string;
+};
+
+export default function useSearchOperationalPoint(props?: SearchOperationalPoint) {
+  const {
+    debounceDelay = 150,
+    mainOperationalPointsOnly = false,
+    searchTerm: initialSearchTerm = '',
+    chCodeFilter: initialChCodeFilter = '',
+  } = props || {};
```

Comment:
use const, export default and rename searchTerm and chCodeFilter in the type
```suggestion
const useSearchOperationalPoint = ({
  debounceDelay = 150,
  mainOperationalPointsOnly = false,
  initialSearchTerm = '',
  initialChCodeFilter = '',
}: SearchOperationalPoint) => {
```